### ISSUE TITLE
Too Many requests

### ISSUE THREAD

**Issue by ale647611** [penguin-teal/gnome-openweather#78]
I received the message that inform me that are many requests of users and suggests me to switch provider settings.
So I try switch to all provider listed, but for each doesn't work.

I think isn't bug , but I think you have scalability problem?

What do you think about?

looks notification and UI of extension

![image](https://github.com/user-attachments/assets/fde13611-14a1-4f9a-bb7a-c8efea4aa6dc)

![image](https://github.com/user-attachments/assets/9b2da3a5-e257-44b2-a6c4-21962ae7fd15)

tks

**Comment by fue36** [comment]
I'll been experiencing the same issue for the past couple of days in US.  Seems like weatherapi.com is blocking connections.

**Comment by ale647611** [comment]
how do I do?

**Comment by ale647611** [comment]
are there alternatives or workaround? I'll must only waiting for restore situation from weathereapi.com side? anybody have same situation now?

**Comment by jasondrusso** [comment]
I have the same issue. Changing providers does not stop the error, and I am no longer able to get weather updates.  These are my settings json (copied from about page):

`{"app-version":"138","git-version":"138.r0.g6367a11","gnome-version":"46.0","user-locale":"en","unit":"'fahrenheit'","custom-keys":["","",""],"appid":"<REDACTED>","weather-provider":"'weatherapicom'","has-run":"true","refresh-interval-current":"3600","locs":[{}],"my-loc-prov":"'ipinfoio'","weatherapidotcom-key":"<REDACTED>","geolocation-appid-mapquest":"<REDACTED>","pressure-unit":"'inHg'","position-in-panel":"'center'","wind-speed-unit":"'mph'"}`

**Comment by fue36** [comment]
The API key was likely revoked.  Can someone can try registering there own API key and placing in the settings in the application under "Personal API Key"?

 How To Get A Weather API Key? 
https://www.youtube.com/watch?v=y3QBwCxfQ6Y

**Comment by Exspiravit** [comment]
> The API key was likely revoked. Can someone can try registering there own API key and placing in the settings in the application under "Personal API Key"?
> 
> How To Get A Weather API Key? https://www.youtube.com/watch?v=y3QBwCxfQ6Y

i get the apikey but how is the option in services? only see WeatherAPI, OpenWeatherMap, visualcrossing and adaptative

**Comment by Exspiravit** [comment]
I have a solution, get apikey from https://openweathermap.org/ and setup in provider section, set provider in openweathermap obviously.

**Comment by fue36** [comment]
✅ In set **Settings** Tab set the **Weather Provider** to **OpenWeather**.  Enable **Use Custom API Key** and for **Personal API Key** use **e54ac00966ee06bcf68722c86925b326**

This is the default weather provider / API Key that the  original OpenWeather Gnome extension used.  This project is a fork of that project.  This workaround will work until the developer is able to update the code.

**Comment by ale647611** [comment]
Thx everybody.

I activated account on openweathermap.org like your suggets , I set api key on the settings tool and now works very well

![image](https://github.com/user-attachments/assets/79b4134b-6604-43b5-a5f2-0de84b24ea6d)

**Comment by MarkyMarkDE** [comment]
Indeed, with the default OpenWeather API-Key e54ac00966ee06bcf68722c86925b326 it worked well again. Seems that all other request methods (without api-key) will be revoked. Also you can use personal a api-key for visualcrossing.com, it is free of use (with some limits) and do work.

**Comment by MarkyMarkDE** [comment]
@penguin-teal this is not a high priority thing, it seems that all included weather services need a personal API-Key so all users must register a free account to get their own key. Without a personal key you always get the message "to many requests, try another [...]" message.
